### PR TITLE
Add a Typescript declaration file

### LINKED ISSUE #6: TypeScript interfaces (definition files)
Could you please add interfaces for TypeScript? Currently everything is undefined/unknown.

### PR DESCRIPTION
**Pull request by Electroid**
Closes #6 by adding a Typescript declaration file.

### CODE REVIEW

**Comment by mhart** [comment]
Oh awesome! Thanks, I'll have a look at this sometime in the next couple of days

File: dist/aws4fetch.d.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+declare module 'aws4fetch' {
```
Comment (blakeembrey) [inline dist/aws4fetch.d.ts]: The wrapping `declare module` is unnecessary for a package definition.

**Comment by Electroid** [comment]
Alright, removed the declaration tag at the top, and rebased to the current master!

**Comment by Electroid** [comment]
@mhart Okay to merge? 😄

**Comment by trjstewart** [comment]
Hey @mhart any chance of getting this merged?

**Comment by blakeembrey** [comment]
@mhart Happy to be a resource to help keep this maintained. I've been using this definition since last last year without issues.

**Comment by sodabrew** [comment]
Likewise, I was excited to see this referenced in the recent Cloudflare blog post, and am using these TypeScript definitions.

**Comment by mhart** [comment]
Hey all – sorry for the radio silence – tough times.

I actually wanted to get this happening with JSDoc comments, but I got sidetracked.

I'll get this in and then move to that – thanks!

**Comment by mhart** [comment]
Released as 1.0.6 – thanks again @Electroid for the contribution 🙌

**Comment by sodabrew** [comment]
Woo! Thank you @mhart!!

**Comment by donocode** [comment]
I might have done something wrong but when I copied these types into my project I had to rename `AwsSigner` to `AwsV4Signer` as that's the name of the exported class

**Comment by mhart** [comment]
I ended up implementing the JSDoc solution and have republished the types in 1.0.7

Let me know if there are any problems!

**Comment by blakeembrey** [comment]
@mhart The new version doesn't work because it's being wrapped in `declare module`, it can't be within a `declare module` to be imported: https://github.com/mhart/aws4fetch/blob/37311c5a39f9bfb07d6223b3cd1605db00067af5/dist/aws4fetch.d.ts#L1

**Comment by mhart** [comment]
Ugh, yeah the module name is a problem. Pity typescript didn't pick that up from any of the test within the repo itself.

Thanks for the PR @blakeembrey – this has been fixed in v1.0.9

**Comment by jivan99** [comment]
@mhart 
The types field in package.json is no longer used if there is an `exports` field. This is causing unresolved type declarations.
[TS Doc Reference](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext)

I have created a PR https://github.com/mhart/aws4fetch/pull/60 to fix the issue.

**Pull request merged**